contracts/core/dexes/CamelotDex.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+// imported contracts and libraries
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+// interfaces
+import "../../interface/ILiquidityDex.sol";
+import "../../interface/camelot/ICamelotRouter.sol";
+
+// libraries
+import "../../libraries/Addresses.sol";
+
+contract CamelotDex is Ownable, ILiquidityDex {
+    using SafeERC20 for IERC20;
+
+    function doSwap(
+        uint256 _sellAmount,
+        uint256 _minBuyAmount,
+        address _receiver,
+        address[] memory _path
+    ) public override returns (uint256) {
+        address sellToken = _path[0];
+
+        IERC20(sellToken).safeIncreaseAllowance(
+            Addresses.camelotRouter,
+            _sellAmount
+        );
+
+        ICamelotRouter(Addresses.camelotRouter)
+            .swapExactTokensForTokensSupportingFeeOnTransferTokens(
+                _sellAmount,
+                _minBuyAmount,
+                _path,
+                _receiver,
+                address(0),
```

Comment:
Do you know anything about how the referral address is used? Might it be good to set it to our multisig for example?

Reply:
Yes! I will handle this